var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -46,6 +47,17 @@ class Protobuf(CMakePackage):
     # first fixed in 3.4.0: https://github.com/google/protobuf/pull/3406
     patch('pkgconfig.patch', when='@:3.3.2')
 
+    def fetch_remote_versions(self):
+        """fix for https://github.com/LLNL/spack/issues/5356"""
+        return dict(map(lambda u:
+                        (u, self.url_for_version(Version(u))),
```

Comment:
This constrains the function to checksum only existing versions in this package. IMO it would be better to call:

`spack.util.web.find_versions_of_archive([], list_url...)`

Or at least I think it will work if archive_urls is an empty list. Then this function should be able to automatically find new releases.

Reply:
thx for the idea, I tried to test that but it did not find version for me.
can you give me a more explicit example, please? I actually want to avoid the scraper in `find_versions_of_archive` since it is:
- not deterministic
- pics a random release if several are found for a version (which will randomly break the hash)

Reply:
ah I think I managed, will push an update! :)